src/components/Secrets/utils/service-account-utils.ts

```diff
@@ -101,3 +116,105 @@ export const unLinkSecretFromServiceAccount = async (secret: SecretKind, namespa
     ],
   });
 };
+
+export const getLinkedServiceAccount = async (secret: SecretKind) => {
+  if (!secret || !secret.metadata?.namespace) {
+    return;
+  }
+  const allServiceAccounts: ServiceAccountKind[] = await K8sListResourceItems<ServiceAccountKind>({
+    model: ServiceAccountModel,
+    queryOptions: {
+      ns: secret.metadata?.namespace,
+    },
+  });
+
+  const linkedServiceAccounts = allServiceAccounts.filter((seviceAccount) =>
+    seviceAccount.secrets.map((linkedSecret) => linkedSecret.name).includes(secret.metadata.name),
+  );
+  const linkedImagePullServiceAccounts = allServiceAccounts.filter((seviceAccount) =>
```

Comment:
one small suggestion here: we could simplify this logic by combining the two filters into one, and also it would avoid potential duplicate service accounts :)

```typescript
const linkedServiceAccounts = allServiceAccounts.filter((sa) => {
  const hasSecret = sa.secrets?.some((s) => s.name === secret.metadata.name);
  const hasImagePullSecret = sa.imagePullSecrets?.some((s) => s.name === secret.metadata.name);
  return hasSecret || hasImagePullSecret;
});
```